pandas/core/algorithms.py

```diff
@@ -2070,20 +2070,38 @@ def sort_mixed(values):
         strs = np.sort(values[str_pos])
         return np.concatenate([nums, np.asarray(strs, dtype=object)])
 
+    def sort_tuples(values: np.ndarray[tuple]):
+        # convert array of tuples (1d) to array or array (2d).
+        # we need to keep the columns separately as they contain different
+        # types and nans (can't use `np.sort` as it may fail when str and nan
+        # are mixed in a column as types cannot be compared).
+        from pandas.core.internals.construction import to_arrays
+        from pandas.core.sorting import lexsort_indexer
+
+        arrays, _ = to_arrays(values, None)
+        indexer = lexsort_indexer(arrays, orders=True)
+        return values[indexer]
+
     sorter = None
+
     if (
         not is_extension_array_dtype(values)
         and lib.infer_dtype(values, skipna=False) == "mixed-integer"
     ):
-        # unorderable in py3 if mixed str/int
         ordered = sort_mixed(values)
     else:
         try:
             sorter = values.argsort()
             ordered = values.take(sorter)
         except TypeError:
             # try this anyway
-            ordered = sort_mixed(values)
+            try:
```

Comment:
ok this looks a lot nicer. can you add some comments on L2093 explaining why we have this nested try / except. (and change the coment on L2097 to be something meaningful.

Reply:
done, I could actually simplify it.